sql-plugin/src/main/scala/com/nvidia/spark/rapids/conditionalExpressions.scala

```diff
@@ -359,15 +362,57 @@ case class GpuCaseWhen(
     if (branchesWithSideEffects) {
       columnarEvalWithSideEffects(batch)
     } else {
-      // `elseRet` will be closed in `computeIfElse`.
-      val elseRet = elseValue
-        .map(_.columnarEvalAny(batch))
-        .getOrElse(GpuScalar(null, branches.last._2.dataType))
-      val any = branches.foldRight[Any](elseRet) {
-        case ((predicateExpr, trueExpr), falseRet) =>
-          computeIfElse(batch, predicateExpr, trueExpr, falseRet)
+      if (caseWhenFuseEnabled && branches.size > 2 &&
```

Comment:
We do not need to evaluate this condition statement for each batch, instead, make it as a lazy val.
```
val (preds, values) = branches.unzip
lazy val canFuse = caseWhenFuseEnabled && branches.size > 2 &&  GpuColumnVectorUtils.isCaseWhenFusionSupportedType(inputTypesForMerging.head) &&
        (values ++ elseValue).forall(_.isInstanceOf[GpuLiteral])

columnarEval (batch: ColumnarBatch): GpuColumnVector = {
  .... 
  if (canFuse) {} else {}
}
```